lib/assignDisabledRanges.js

```diff
@@ -66,35 +66,53 @@ module.exports = function (root, result) {
 		if (inlineEnd) {
 			// Ignore comments already processed by grouping with a previous one.
 			if (inlineEnd === comment) inlineEnd = null;
-		} else if (isInlineComment(comment)) {
-			const fullComment = comment.clone();
-			let next = comment.next();
-			let lastLine = (comment.source && comment.source.end && comment.source.end.line) || 0;
 
-			while (next && next.type === 'comment') {
-				/** @type {PostcssComment} */
-				const current = next;
+			return;
+		}
 
-				if (!isInlineComment(current)) break;
+		const next = comment.next();
+
+		// If any of these conditions are not met, do not merge comments.
+		if (
+			!(
+				isInlineComment(comment) &&
+				comment.text.startsWith('stylelint-') &&
```

Comment:
This will match commits with `stylelint-whatever`. Users could put stylelint plugins names in comments. I think we should change comments starting with `stylelint-disable` and `stylelint-enable` to eliminate problems.

Same applies for a `stylelint-` check below.